internal/api/handlers.go

```diff
@@ -125,7 +130,7 @@ func GetCertificateRequest(env *Environment) http.HandlerFunc {
 		cert, err := env.DB.RetrieveCSR(id)
 		if err != nil {
 			if err.Error() == "csr id not found" {
-				logErrorAndWriteResponse(err.Error(), http.StatusBadRequest, w)
+				logErrorAndWriteResponse(err.Error(), http.StatusNotFound, w)
```

Comment:
I'm still not sure if we should use a 404 here. It may indicate to users that the whole URL is wrong instead of just the resource being missing. But at the same time a lot of websites do exactly that.

Reply:
Yeah, I tried to do some research and 404 seemed the most appropriate for me, plus the error message `user id not found` is clear I think.

Reply:
sure lets keep it